samples/joke/README.md

```diff
@@ -47,11 +47,128 @@ cluster. You can just follow the steps below to get started quickly:
+So far, we have built the Operator bundle that is ready to be installed on a cluster, but it's not registered in any catalog yet. For doing this, the `olm` tool provides a utility to add new Operator bundle images to an existing catalog by doing:
+
+```shell
+opm index add --bundles $BUNDLE_IMAGE --tag $INDEX_IMAGE --build-tool docker --skip-tls
+          
+          
+opm index add \
+    --bundles <your container registry>/<your container registry namespace>/joke-manifest-bundle:<tag> \
+    --tag <catalog container registry>/<catalog container registry namespace>/<catalog name>:<tag> \
+    --build-tool docker --skip-tls
+docker push <catalog container registry>/<catalog container registry namespace>/<catalog name>:<tag>
+```
+
+| If you're using an insecure registry, you'd need to append the argument `--skip-tls` to the `opm index` command.
+
+The `<catalog container registry>/<catalog container registry namespace>/<catalog name>:<tag>` image is the catalog image that must be registered on the cluster. 
+If you want to register a custom catalog, you can create it and configure it with the catalog image by doing the next command:
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: joke-operator
+  namespace: operators
+spec:
+  sourceType: grpc
+  image: <catalog container registry>/<catalog container registry namespace>/<catalog name>:<tag>
+EOF
+```
+
+Once the catalog is installed, you should see the catalog pod up and running:
+
+```shell
+kubectl get pods -n operators --selector=olm.catalogSource=joke-operator
+```
+
+4. Install your operator via OLM
+
+OLM deploys operators via [subscriptions](https://olm.operatorframework.io/docs/tasks/install-operator-with-olm/#install-your-operator). You can simply create the `Subscription` resource that contains the operator name and channel to install by running the following command:
+
+```shell
+cat <<EOF | kubectl create -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: joke-subscription
+  namespace: <your namespace>
+spec:
+  channel: alpha
+  name: joke
+  source: joke-operator
+  sourceNamespace: operators
+EOF
+```
+
+Once the subscription is installed, you should see your operator pod up and running:
+
+```shell
+kubectl get csv -n operators jokerequestreconciler
+```
+
+Also, this example needs the `Joke` CRD to be installed on the cluster:
+
+```shell
+kubectl apply -f classes/META-INF/fabric8/jokes.samples.javaoperatorsdk.io-v1.yml -n <your namespace>
+```
+
+5. Test your operator
+
+Deploy the test request (or your own): `kubectl apply -n <your namespace> -f src/main/resources/jokerequest.yml` and,
+if everything went well, a `Joke` resource named after the `id` of the joke retrieved from the API will be
+  created on your cluster: `kubectl get jokes -n <your namespace>`.
```

Comment:
```suggestion
As mentioned at the beginning of this document, the idea is that you create `JokeRequest` Custom Resources in the cluster. Let's now test if we can do it via the operator that we have just deployed. This can be achieved by running the `kubectl apply -n <your namespace> -f src/main/resources/jokerequest.yml` command. Make sure you change the file to apply with your own.
If everything went well, a `JokeRequest` resource named after the `id` of the joke retrieved from the API will be created on your cluster, you can check it with the `kubectl get jokes -n <your namespace>` command.
 
```

Reply:
I propose this wording that makes things clearer for me. At the moment where I'm doing this review I'm a little bit confusing about the `Joke` and `JokeRequest` so the wording might be wrong. From what I have understood, the operator should create custom resources of type `JokeRequest`, the `Joke` resources are also created but as a dependency, I mean the `Joke` resource are not the inputs to the operator. Could you please confirm this @metacosm @Sgitario ?

Reply:
No, the JokeRequest resource is the request from users. They need to create it by doing the `kubectl create -f ...`. The operator will create the Joke resource with the response of the user request.